src/components/TextInput/TextInput.tsx

```diff
@@ -20,10 +20,13 @@ export type TextInputProps = NativeAttributes<'input'> & {
 
   /** The icon present on the input  */
   icon?: IconProps['type'];
+
+  /** Additional icon props  */
+  iconProps?: Omit<IconProps, 'type' | 'ref'>;
```

Comment:
Hmm I feel that `IconProps` is a non-restricting enough. Currently  I can add `iconProps={{ position: 'fixed' }}` and totally mess with your layout  😄 

Should we perhaps either allow a fixed set of icon props or split them into separate ones `i.e. `iconSize`, `iconColor` instead of a single `iconProps`)

Reply:
And it's done!